response.go

```diff
@@ -441,6 +441,13 @@ func visitModelNode(model interface{}, included *map[string]*Node,
 					}
 				}
 			}
+		} else if annotation == annotationLinks {
+			// This is for marshalling.
+			// This is left blank intentionally as the handling
+			// of `jsonapi:"links"` is done below via the
+			// Linkable interface.
+			// And the logic for Marshalling links should be done
+			// via that interface.
```

Comment:
This function is called during [MarshalPayload](https://github.com/hashicorp/jsonapi/blob/master/response.go#L65-L77). The links functionality here is handled through the [Linkable interface](https://github.com/hashicorp/jsonapi/blob/master/node.go#L96-L98). We don't want to do anything here because if the resource struct implements the interface, then links will be handled there. Otherwise, we just want to handle the annotaiton or else it will [blow up below](https://github.com/hashicorp/jsonapi/pull/4/files#diff-6b94feb58536b61f1aaf93471d52ee58748d9252f08e12cd9cce2feb44c49691R453).